render/mapping.go

```diff
@@ -49,6 +32,88 @@ type MapFunc func(report.NodeMetadata) (RenderableNode, bool)
 // node IDs prior to mapping.
 type PseudoFunc func(srcNodeID string, srcNode RenderableNode, dstNodeID string) (RenderableNode, bool)
 
+// MapFunc is anything which can take an arbitrary RenderableNode and
+// return another RenderableNode.
+type MapFunc func(RenderableNode) (RenderableNode, bool)
+
+// MapEndpointIdentity maps a endpoint topology node to endpoint RenderableNode node.
+// As it is only ever run on endpoint topology nodes, we can safely assume the
+// presences of certain keys.
+func MapEndpointIdentity(m report.NodeMetadata) (RenderableNode, bool) {
+	var (
+		id    = fmt.Sprintf("endpoint:%s:%s:%s", m[report.HostNodeID], m["addr"], m["port"])
+		major = fmt.Sprintf("%s:%s", m["addr"], m["port"])
+		minor = fmt.Sprintf("%s (%s)", m[report.HostNodeID], m["pid"])
+		rank  = m["pid"]
+	)
```

Comment:
Ah but you can, as this mapper is only every run on the endpoints topology, which always contains a pid.